src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -508,36 +508,99 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
-               int64_t& head_size,
-               int64_t& block_size) {
-                if (bychannel) {
-                    // TODO: need to handle group size != block size case
-                    if (precision == ov::element::i8 || precision == ov::element::u8) {
-                        block_size += infer_precision.size() * 2;
-                    }
-                } else {
-                    if (precision == ov::element::i8 || precision == ov::element::u8) {
-                        head_size += infer_precision.size() * 2 * group_num;
-                    }
+        {
+            // Disable XAttention if GPU Xe2/Xe3 architectures is unavaiable or IGC incompatiable.
+            auto check_xattn_gpu_compatibility  = [&](void) -> bool {
+                        auto& engine = m_context->get_engine();
+                        const auto& info = engine.get_device_info();
+                        if (info.arch != cldnn::gpu_arch::xe2 && info.arch != cldnn::gpu_arch::xe3) { // CM optimized for systolic-array architectures
+                            return false;
+                        }
+
+#ifdef GPU_DEBUG_CONFIG
+                        if (!config.get_use_cm()) {
+                            OPENVINO_WARN("You may miss XAttention optimization,"
+                                        "as CM for usage is disabled. Enable it by setting environment variable OV_GPU_USE_CM=ON.");
+                            return false;
+                        }
+#endif
+
+                        if (!check_cm_jit_support(engine, config)) {
+                            OPENVINO_WARN("You may miss XAttention optimization,"
+                                        "as current IGC version is not compatible to the CM kernel used. Enable it by updating IGC."
+                                        "Please also make sure clangFEWrapper for CM is present by checking environment varibles like "
+                                        "CM_FE_DIR or LD_LIBRARY_PATH if you are using Linux.");
+                            return false;
+                        }
+
+                        return true;
+                    };
+
+            // Determine if XAttention is enabled by user (via GENAI) by checking if model parameters contains
+            // xattention configurations, which are added in SDPAToPagedAttention pass.
+            bool use_xattention = false;
+            const auto& parameters = func->get_parameters();
+            for (const auto& param : parameters) {
+                if (param->get_friendly_name() == "xattention_block_size") {
+                    use_xattention = true;
+                    break;
                 }
-            });
+            }
+
+            if (use_xattention) {
+                // Throw exception if xattn is not supported by either GPU archieture or compiler.
+                OPENVINO_ASSERT(check_xattn_gpu_compatibility(), "XAttention is not supported by either GPU archieture or IGC you are using.");
```

Comment:
Please use OPENVINO_THROW to be catched becasue assertion is not terminating in release build

Reply:
done.